bmm/src/main/java/org/openehr/bmm/v2/persistence/PBmmClass.java

```diff
@@ -125,12 +164,76 @@ public void setSourceSchemaId(String sourceSchemaId) {
         this.sourceSchemaId = sourceSchemaId;
     }
 
-    public Map<String, PBmmType> getAncestorDefs() {
+    public Map<String, PBmmUnitaryType> getAncestorDefs() {
         return ancestorDefs;
     }
 
-    public void setAncestorDefs(Map<String, PBmmType> ancestorDefs) {
-        this.ancestorDefs = ancestorDefs;
+    public BmmClass createBmmClass() {
+        BmmClass bmmClass;
+        if (getGenericParameterDefs().size() > 0) {
+            bmmClass = new BmmGenericClass(getName(), getDocumentation(), nullToFalse(isAbstract()));
+        } else {
+            bmmClass = new BmmSimpleClass(getName(), getDocumentation(), nullToFalse(isAbstract()));
+        }
+        bmmClass.setOverride(nullToFalse(isOverride()));
+        bmmClass.setSourceSchemaId(getSourceSchemaId());
+        return bmmClass;
+    }
+
+    public BmmClass populateBmmClass(BmmClassProcessor bmmModel, PBmmSchema schema) {
+        BmmClass bmmClass = bmmModel.getUnprocessedClassDefinition(getName());
+        if (bmmClass != null) {
+            // populate references to ancestor classes; should be every class except Any
+            BmmType bmmType;
+            for (PBmmUnitaryType ancestorType : ancestorRefs(schema).values()) {
+                BmmClass ancestorClass = bmmModel.getClassDefinition(ancestorType.baseType());
+                if (ancestorClass != null) {
+                     bmmType = ancestorType.createBmmType(bmmModel, ancestorClass);
+                } else {
+                    throw new RuntimeException("Error retrieving class definition for ancestor class " +
+                            ancestorType.baseType() + " of BmmClass " + name);
+                }
+
+                if (bmmType instanceof BmmDefinedType) {
+                    bmmClass.addAncestor((BmmDefinedType) bmmType);
+                } else {
+                    throw new RuntimeException("Ancestor class definition of the wrong type for ancestor class " +
+                            ancestorType.baseType() + " of BmmClass " + name);
+                }
+            }
+
+            // create generic parameters if a generic class
+            // and add to the BMM_GENERIC_TYPE.generic_parameters list
+            if (bmmClass instanceof BmmGenericClass) {
+                for (PBmmGenericParameter param : getGenericParameterDefs().values()) {
+                    BmmParameterType bmmGenericParameter = param.createBmmGenericParameter(bmmModel);
+                    if (bmmGenericParameter != null) {
+                        ((BmmGenericClass) bmmClass).addGenericParameter(bmmGenericParameter);
+                    } else {
+                        throw new RuntimeException("Error retrieving class definition for generic parameter " +
+                                param.getName() + " of PBmmClass " + name);
+                    }
```

Comment:
`createBmmGenericParameter` always returns a non-null result. The if-statement can be removed.
```suggestion
                    BmmParameterType bmmGenericParameter = param.createBmmGenericParameter(bmmModel);
                    ((BmmGenericClass) bmmClass).addGenericParameter(bmmGenericParameter);
```